dnYara/CompiledRules.cs

```diff
@@ -11,17 +13,25 @@ public sealed class CompiledRules
         : IDisposable
     {
         internal IntPtr BasePtr { get; set; }
+        private YR_RULES _struct = default(YR_RULES);
```

Comment:
This private variable is only needed very temporarily, so I wonder if it is necessary to make it global to the class? This may result in memory leakage (because the structure references other structures) if the class is poorly maintained / disposed. Moreover, using the "pointer only" instead of the struct value allows to remain neutral on the content of the structure in case of YARA evolution.

We may even be able to suppress the GetStruct() function, which seems useless in all cases of use.

Finally, the complete YR_RULES structure could be completely removed from the project because only its pointer is enough.

Reply:
This is a good point. We can marshal in the constructor to get the offsets we need to create the Rules lists and such, but no other code should need the offset.

Reply:
I've removed this now.